### DISCUSSION TITLE
Temporary skip on scan?

### DISCUSSION THREAD

**Discussion by jclagreca** [egzumer/uv-k5-firmware-custom#169] [Ideas]
When scanning all memory banks, it would be nice to be able to temporarily skip a specific memory for the duration of that scan.  I know I can manually skip over a memory with the arrow buttons, but it's annoying to have to do that every loop.  Would be nice if we could press a button that would skip that particular memory for the duration of the scan.

**Comment by ErikS-web** [comment]
That option can be done with Removing that memory-channnel from the scanlist you are scanning.

See the manual under [Memory-channels-scanning](https://github.com/egzumer/uv-k5-firmware-custom/wiki/Radio-operation#memory-channels-scanning)

pse confirm.

**Comment by jclagreca** [comment]
Understood, but still not quite the same feature.  If you remove the channel from the scan, you have to remember to add it back in, which will most likely be forgotten.  

Would be nice i we could long press the arrow button, or some other button that then excludes the channel ONLY for the duration of that scan.  This is useful if there is a long conversation happening on a channel that you normally scan, but are not interested in it at the moment.   That way, in the future, the channel will still be included in your scan bank, or also would be useful when scanning ALL channels.  

One specific example would be if I have NOAA frequencies programmed to memory.  EVERY single loop scan of ALL memories will stop on the local NOAA station.  Would be nice if I could just exclude it on the first time around, and then continue scanning through all channels.

**Comment by ErikS-web** [comment]
Read the manual: `*` star means: scan All memory channels, not only I or II.

You can always  place a skipped channel back into the scanlist with long press `5 NOAA` button,
Or relead from PC.

**Reply by jclagreca** [reply]
I have read the manual and understand how it works.  

If you are scanning a group of channels, and one of them is busy with something you aren't interested in, or noise, it makes MUCH MORE SENSE to temporarily skip that channel for the duration of the scan, rather than try to remove and then remember to add it back in later.  Otherwise, what if you run into 2 channels that you want to skip, now you have to stop the scan, remove the channels, and start the scan again?  Makes much more sense to press a single button to remove the channel ONLY for the duration of the scan.

**Comment by f5gkw** [comment]
Ijv firm does that,  temporary exclusion by pressing * during the scan.
That doesn't delete the frequency from the memory or scanlist.

**Reply by ErikS-web** [reply]
Where do they share the code ?

**Reply by DualTachyon** [reply]
They don't...

**Comment by f5gkw** [comment]
I don't know if the code is somewhere.

The firmwares ( and yours in other folder) are here
https://github.com/spm81/Quansheng_UV-K5/tree/main/Firmware%2FUNIVERSAL%20FIRMWARE%20(%20All%20Radios%20-%20UV-K5%2C%20UV-K5(8)%2C%20UV-K6%2C%20UV-5R%20Plus%2C%20etc...)%2FCustom%20Firmware%20by%20IJV%20(Indiajulietvictor)

**Comment by f5gkw** [comment]
A link via telegram, about the latest ujv version ( less than 1 hour)
https://t.me/quansheng_firmware/75/2656

Le ven. 1 déc. 2023 à 22:51, Jean-Philippe Buchet ***@***.***> a
écrit :

**Comment by f5gkw** [comment]
You can scan between 2 frequencies and use the * to skip the annoying
frequency -ies you have
I didn't test the * feature in this case or in memories scanlist.

**Comment by f5gkw** [comment]
I quickly test on scanlist, you don't need to stop the scan to blackist a
memory, just press on *.

**Comment by jclagreca** [comment]
I was able to find the feature and instructions on how to use it here:  
[https://www.universirius.com/preppers/quansheng-uv-k5-manuale-del-firmware-ijv/](https://www.universirius.com/preppers/quansheng-uv-k5-manuale-del-firmware-ijv/)

In the 7.1 Scanning section of the page (translated to english):

>  Black List:
> You can exclude up to 40 unwanted frequencies in a Black List. This only works for VFO scans.
> When the scan stops on an unwanted frequency, briefly press the *️⃣S can button, it will be added to the Black List. The writing: “BlackList In #(n)” will appear on the screen where (n) is the number of frequencies entered.
> Exiting scan mode or turning off the radio will clear the Black List.

This is the person who makes the LJV firmware:  [https://www.qrz.com/db/IU0IJV](https://www.qrz.com/db/IU0IJV) and he can be contacted via email here:  [iu0ijv@domox.org](mailto:iu0ijv@domox.org)

Don't get me wrong, egzumer firmware is excellent, but it would be nice if this feature were also included!

**Comment by f5gkw** [comment]
Yep.
But this page is based on a previous version.
The V2.3R2 is just released.
If you have  a telegram account :
https://t.me/quansheng_firmware/75/2672

Since at least v2.1, the blacklist with * key works on both memories scan
or vfo scan.
73, F5GKW

**Comment by majonezz** [comment]
I think this is a good idea. I have this in my Baofeng with OpenGD77. Temporarily skip unwanted scan result by pressing one button. If the author really needs that i implemented this in my fork, but my fork may be not up to date, so some effort may be needed to resync it with current files in this repo. 
https://github.com/egzumer/uv-k5-firmware-custom/blob/876d1289e97a4ea92444652ec0a57c2d0a109cfd/app/chFrScanner.c#L37-L59
https://github.com/egzumer/uv-k5-firmware-custom/blob/876d1289e97a4ea92444652ec0a57c2d0a109cfd/app/chFrScanner.c#L171-L205
https://github.com/egzumer/uv-k5-firmware-custom/blob/876d1289e97a4ea92444652ec0a57c2d0a109cfd/app/chFrScanner.c#L280-L288
https://github.com/egzumer/uv-k5-firmware-custom/blob/876d1289e97a4ea92444652ec0a57c2d0a109cfd/app/action.c#L311-L320

**Comment by tjcirillo** [comment]
i agree with the OP. i posted this desire elsewhere and got nowhere.  if you're scanning memory channels (not frequencies) it doesn't appear possible to temporarily block one until the next scan, unless i'm wrong

**Comment by JBryant915** [comment]
Love my Quansheng, but coming from Baofeng, I miss this feature. Could temp skip a mem channel by using the * key or could use chirp to set certain ones to skip. With the quansheng, I can't find a way to temp skip a mem channel, and the option to skip in chirp is not there either. I would love to see this feature myself. Yes, these frequencies can be deleted and added easily, but this feature would be way easier.

**Comment by tjcirillo** [comment]
Me too!

**Comment by jan-Sanku** [comment]
This would be quite nice! The spectrum analyzer already has this feature, I think it'd make just as much sense here.

Reminder to everyone commenting they want this: there's an upvote button!

**Comment by AubsUK** [comment]
I like this idea and think it's something I could benefit from, and would be an enhancement to what I've already done (everything scanning related!).  I've just finished the first version of firmware and CHIRP driver over on my repo - https://github.com/AubsUK/uv-k5-firmware-custom

The current main changes to Egzumer's firmware are:

* The two ScanLists have been removed and replaced with 10 ScanLists (0-9)
* When scanning, the ScanLists now show along the top of the screen
* Toggling ScanLists on/off is as simple as pressing the number
* Memory channels can now be excluded (LockOut/Skip) from scanning
* Scanning can be set to automatically start when the radio is turned on
* All the changes can be managed through CHIRP

You WILL need a custom CHIRP driver to use CHIRP.  All instructions provided.

Open a discussion there if you have any questions or suggestions.

**Reply by AubsUK** [reply]
I've just added this feature to my repo, and I'm already using it! Thank you for suggesting it :-)
